### DISCUSSION TITLE
tausurface mode

### DISCUSSION THREAD

**Discussion by seanandrews** [dullemond/radmc3d-2.0#29] [General]
Hi @dullemond,

I'm trying to understand the outputs in 'tausurface_3d.out' generated when using a command like
`radmc3d tausurf 1.0 loadlambda 10 incl 45 posang 30 npix 512`.  I think I understand the format: it gives me 
(x, y, z) locations for the tau = 1.0 surface in an nlam, npix, npix grid.  But I guess I don't understand what coordinate frame (x, y, z) are in.  If they're sky-projected, I'd expect (x, y) for a given pixel to be the same at all frequencies (only z would change).  Are (x, y, z) in the disk reference frame?  Do they account for the incl, posang projections?

thanks!
Sean

**Comment by dullemond** [comment]
Dear Sean,

I hope I understand your question correctly (and I hope I remember it myself correctly). If I am not mistaken: It depends on your inclination. If you would set incl 0 and posang 0, then the xyz of the image are the xyz of the 3D box. The image values are then the z coordinate. But if you incline it, then the x and y of the image are not the same as the x and y of the 3D box. They are the x and y of the image plane (but of course still in cm). The image value itself is the distance from the plane that goes through (0,0,0) but that is rotated by incl and posang. It is, in a way, as if you simply rotate the box. 

Does this answer your question?

Cheers,
Kees

PS: I am not sure if it works properly for disk models that only have the upper half (i.e. theta<=pi/2) and use  mirroring in the midplane.

**Comment by seanandrews** [comment]
Hi Kees,

I'm still confused I guess.  The 'tausurface3d.out' gives me a cube (nlam, npix, npix) of (x, y, z) coordinates: call it `taulocs`.  If I go to some specific location in the cube, say `taulocs[10, 200, 100]`, don't I already know the sky-projected location (x_sky, y_sky) of that cell based on the (specified) pixel scale?  If the returned (x, y, z) coordinates in that location are sky-projected offsets from the image center, then I would expect x = x_sky and y = y_sky for all wavelengths.  i.e., if the (x, y) are sky-projected then their location in the cube and these entries in the coordinates are redundant.  This is not the case (!), so that makes me suspect that the (x, y) are not sky-projected offsets, but rather in the object/structure(/disk) reference frame.  

Now, if I'm understanding you correctly, you are effectively saying that the (x, y, z) are in the reference frame of the "3D box", which I assume to mean the object/structure(/disk) frame.  Is that correct?  In that case, there'd be a straightforward (modulo getting the directionalities right) coordinate transformation between the sky-projected (x_sky, y_sky) pixel locations and the (x, y, z) outputs for a given incl and posang....right?  

For reference, I'm trying to ascertain the effective photosphere layer for CO spectral line models of disks.  But indeed, I'm using a midplane mirroring so maybe this is all moot (although perhaps its ok, since tau should be very high and the near surface should be well above the midplane?)

thanks!
Sean

**Comment by dullemond** [comment]
Dear Sean,

ah, I see my confusion. So what I usually do is to look at the image.out. That gives the distance along the line of sight. 
> I'm still confused I guess. The 'tausurface3d.out' gives me a cube (nlam, npix, npix) of (x, y, z) coordinates: call it taulocs. If I go to some specific location in the cube, say taulocs[10, 200, 100], don't I already know the sky-projected location (x_sky, y_sky) of that cell based on the (specified) pixel scale?
> 
Yes, correct. 
> If the returned (x, y, z) coordinates in that location are sky-projected offsets from the image center,
> 
No, I was confused with the image.out. That gives the offset.
> then I would expect x = x_sky and y = y_sky for all wavelengths. i.e., if the (x, y) are sky-projected then their location in the cube and these entries in the coordinates are redundant. This is not the case (!), so that makes me suspect that the (x, y) are not sky-projected offsets, but rather in the object/structure(/disk) reference frame.
> 
Yes! Correct.
> Now, if I'm understanding you correctly, you are effectively saying that the (x, y, z) are in the reference frame of the "3D box", which I assume to mean the object/structure(/disk) frame. Is that correct?
> 
Yes.
> In that case, there'd be a straightforward (modulo getting the directionalities right) coordinate transformation between the sky-projected (x_sky, y_sky) pixel locations and the (x, y, z) outputs for a given incl and posang….right?
> 
Yes. 
> For reference, I'm trying to ascertain the effective photosphere layer for CO spectral line models of disks. But indeed, I'm using a midplane mirroring so maybe this is all moot (although perhaps its ok, since tau should be very high and the near surface should be well above the midplane?)
> 
Just try to extend toward the other side (in theta). But I am not sure if that is necessary. I just seem to remember it.

Best wishes,
Kees

**Reply by seanandrews** [reply]
Thanks, Kees....I can confirm that this did indeed work great!  I appreciate the quick replies and all the help.

best,
Sean
